### PR TITLE
Moved url validation to validation.js

### PR DESCRIPTION
**Pull request by sarthak-marwaha**

### CODE REVIEW

File: src/foam/core/Validation.js [hunk 1]
```diff
@@ -541,3 +541,26 @@ foam.CLASS({
...
   ]
 });
+
+foam.CLASS({
+  package: 'foam.core',
+  name: 'URLValidationRefinement',
+  refines: 'foam.core.URL',
+
+  messages: [
+    { name: 'INVALID_WEBSITE', message: 'Invalid website' }
```
Comment (kgrgreer) [inline src/foam/core/Validation.js:551]: If it is a URL, then do you know it's a website?

File: src/foam/core/Validation.js [hunk 2]
```diff
@@ -541,3 +541,26 @@ foam.CLASS({
     }
   ]
 });
+
+foam.CLASS({
```
Comment (kgrgreer) [inline src/foam/core/Validation.js:545]: Should you also remove the validation from User.js and elsewhere as part of this PR?

Comment (sarthak-marwaha) [reply src/foam/core/Validation.js:545]: I did remove it. Not sure what you mean

File: src/foam/core/Validation.js [hunk 3]
```diff
@@ -541,3 +541,71 @@ foam.CLASS({
...
+  ],
+
+  properties: [
+    {
+      class: 'FObjectArray',
+      of: 'foam.core.ValidationPredicate',
+      name: 'validationPredicates',
+      factory: function() {
+        var self = this;
+        var websiteRegex = /(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9]\.[^\s]{2,})/;
```
Comment (kgrgreer) [inline src/foam/core/Validation.js:595]: Does it accept http also?

**Pull request merged**